src/tidecli/utils/file_handler.py

```diff
@@ -120,33 +122,47 @@ def create_task(task: TaskData, overwrite: bool, user_path: str | None = None) -
     """
     # Sets path to current path or user given path
     if user_path:
-        user_folder = Path.cwd() / user_path
+        user_folder_root = Path.cwd() / user_path
     else:
-        user_folder = Path.cwd()
+        user_folder_root = Path.cwd()
 
     # Add course path to create task path
-    user_folder = user_folder / Path(task.path).name / task.ide_task_id
+    if task.task_directory is not None:
+        user_folder = user_folder_root / task.task_directory
+    else:
+        user_folder = user_folder_root / Path(task.path).name / task.ide_task_id
 
     # Fix for tasks that have file_name without suffix
+    # TODO: Why this, why there can not be files without suffix?
     for f in task.task_files:
         path = Path(f.file_name)
         suffix = path.suffix
-        if suffix != "":
-            continue
-        f.file_name = add_suffix(f.file_name, task.run_type)
+        if suffix == "":
+            f.file_name = add_suffix(f.file_name, task.run_type)
+        if f.task_directory:
+            f.saved_full_path = str(user_folder_root / f.task_directory / f.file_name)
+        else:
+            f.saved_full_path = str(user_folder / f.file_name)
 
     saved = save_files(
         task_files=task.task_files, save_path=user_folder, overwrite=overwrite
     )
- 
+
     if task.supplementary_files is not None:
-        save_files(task_files=task.supplementary_files, save_path=user_folder, overwrite=overwrite)
+        for f in task.supplementary_files:
+            if f.task_directory:
+                f.saved_full_path = str(user_folder_root / f.task_directory / f.file_name)
+            else:
+                f.saved_full_path = str(user_folder / f.file_name)
```

Comment:
The save path logic is now in two places, here and on line 207. Is this necessary? If not, it'd be better to refactor to a single place.

Reply:
The whole logic is changed and is moved to save_files